packages/registry/contracts/CannonRegistry.sol

```diff
@@ -293,19 +276,26 @@ contract CannonRegistry is EfficientStorage, OwnedUpgradable {
     return v;
   }
 
+  function _canPublishPackage(Package storage _package, address _publisher) internal view returns (bool) {
+    address owner = _package.owner;
+
+    if (owner == _publisher) return true;
+
+    uint256 additionalDeployersLength = _package.additionalDeployersLength;
+    for (uint256 i = 0; i < additionalDeployersLength; i++) {
+      if (_package.additionalDeployers[i] == _publisher) return true;
+    }
+
+    return false;
+  }
+
   function _checkCrossDomainSender() internal {
     // we can only receive change ownership requests from our counterpart on mainnnet
     if (_OPTIMISM_RECEIVER.xDomainMessageSender() != address(this)) {
       revert Unauthorized();
     }
   }
 
-  function _propogatePackageOwner(bytes32 _packageName, address _owner, address[] memory _additionalDeployers) internal {
```

Comment:
Why did we remove this?
We just deployed a version of the CannonRegistry so this will break the partial deployment i believe